compiler/coreSyn/CoreLint.hs

```diff
@@ -1225,7 +1225,13 @@ lintCoreAlt lookup_scrut scrut_ty scrut_weight alt_ty alt@(DataAlt con, args, rh
         -- type variables of the data constructor
         -- We've already check
       lintL (tycon == dataConTyCon con) (mkBadConMsg tycon con)
-    ; let { con_payload_ty = piResultTys (dataConRepType con) tycon_arg_tys }
+    ; let { con_payload_ty = piResultTys (dataConRepType con) tycon_arg_tys
+          ; (_, pred, con_args, _) = dataConSig con
+          ; ex_tvs = dataConExTyVars con
+          -- Weights of the things that the data con brings into scope
+          -- MattP: TODO: Move this into DataCon
+          ; weights = (replicate (length ex_tvs + length pred) Omega)
```

Comment:
Is the `Omega`  here intentional?

Reply:
Yes, that's the weight of the type and dictionary arguments which are always `Omega`.